tests/testthat/test-migrate.R

```diff
@@ -388,3 +388,305 @@ test_that("migrate() coerces 'character'-type `state` columns to type 'factor'",
+
+  expect_error(
+    migrate(
+      data = mock_credit_with_missing,
+      time = date,
+      state = risk_rating,
+      id = customer_id,
+      percent = FALSE,
+      fill_state = c("NR", "No Rating"),
+      verbose = FALSE
+    ),
+    reg_exp = "`fill_state` argument must be length-one value (of type `character`, `numeric`, or `factor`)"
+  )
+
+})
+
+test_that("migrate() errors when `fill_state` is not of type `character`, `numeric` or `factor`", {
+
+  # Logical
+  expect_error(
+    migrate(
+      data = mock_credit_with_missing,
+      time = date,
+      state = risk_rating,
+      id = customer_id,
+      percent = FALSE,
+      fill_state = FALSE,
+      verbose = FALSE
+    ),
+    reg_exp = "`fill_state` argument must be length-one value (of type `character`, `numeric`, or `factor`)"
+  )
+
+  # Dataframe
+  expect_error(
+    migrate(
+      data = mock_credit_with_missing,
+      time = date,
+      state = risk_rating,
+      id = customer_id,
+      percent = FALSE,
+      fill_state = iris,
+      verbose = FALSE
+    ),
+    reg_exp = "`fill_state` argument must be length-one value (of type `character`, `numeric`, or `factor`)"
+  )
+
+})
+
+test_that("migrate() works when `fill_state` is of type `character`", {
+
+  migrated_data <- migrate(
+    data = mock_credit_with_missing,
+    time = date,
+    state = risk_rating,
+    id = customer_id,
+    percent = FALSE,
+    fill_state = "NR",
+    verbose = FALSE
+  )
+
+  expect_true(
+    inherits(migrated_data, "data.frame")
+  )
+
+})
+
+test_that("migrate() works when `fill_state` is of type `numeric`", {
+
+  migrated_data <- migrate(
+    data = mock_credit_with_missing,
+    time = date,
+    state = risk_rating,
+    id = customer_id,
+    percent = FALSE,
+    fill_state = 999,
+    verbose = FALSE
+  )
+
+  expect_true(
+    inherits(migrated_data, "data.frame")
+  )
+
+})
+
+test_that("migrate() works when `fill_state` is of type `factor`", {
+
+  migrated_data <- migrate(
+    data = mock_credit_with_missing,
+    time = date,
+    state = risk_rating,
+    id = customer_id,
+    percent = FALSE,
+    fill_state = as.factor("No Rating"),
+    verbose = FALSE
+  )
+
+  expect_true(
+    inherits(migrated_data, "data.frame")
+  )
+
+})
+
+test_that("migrate() works when `fill_state` is a value that already exists", {
+
+  migrated_data_without_missing <- migrate(
+    data = mock_credit,
+    time = date,
+    state = risk_rating,
+    id = customer_id,
+    percent = FALSE,
+    verbose = FALSE
+  )
+
+  migrated_data_with_missing <- migrate(
+    data = mock_credit_with_missing,
+    time = date,
+    state = risk_rating,
+    id = customer_id,
+    percent = FALSE,
+    fill_state = "CCC",
+    verbose = FALSE
+  )
+
+  expect_identical(nrow(migrated_data_without_missing), nrow(migrated_data_with_missing))
+  expect_identical(sum(migrated_data_without_missing$count),sum(migrated_data_with_missing$count))
+
+})
```

Comment:
This is a great test, and I hadn't even thought of this scenario / use case.  I think this is the correct behavior, but I wonder if we should inform/warn the user that they are mapping observations with missing beginning/end states to an *existing* class, instead of creating a new class.  What do you think @IvanM26 ?

Reply:
Perhaps we can create a message that says something like:

"Migrating missing timepoints into *new* class 'NR' " or  "Migrating missing timepoints into *existing* class 'ABC' ".  

This message would only appear if `verbose = TRUE`.

Reply:
Great idea! Messages have been created and new unit test added!